### PR TITLE
add utility to check if char is inside string

### LINKED ISSUE #17: Write a small utility to check if a match is inside a string
what it says on the tin

### PR DESCRIPTION
**Pull request by MarcusWichelmann**
fixes #17 
check if the char at the given index is inside a string (or char).

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 vala_precompile (LIB_VALA_C
     Linter.vala
     Check.vala
+    Utils.vala
```
Comment (Waishon) [inline lib/CMakeLists.txt]: No alphabetical order here?

File: lib/Utils.vala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            c \ => is_in_string = true, is_in_char = false, escape_on = true
+              " => is_in_string = true, is_in_char = false, escape_on = false
+            d ' => is_in_string = true, is_in_char = false, escape_on = false
+            e ' => is_in_string = true, is_in_char = false, escape_on = false
+            f \ => is_in_string = true, is_in_char = false, escape_on = true
+              " => is_in_string = true, is_in_char = false, escape_on = false
+            g " => is_in_string = false, is_in_char = false, escape_on = false
+            h ' => is_in_string = false, is_in_char = true, escape_on = false
+            i ' => is_in_string = false, is_in_char = false, escape_on = false
+        */
```
Comment (Waishon) [inline lib/Utils.vala]: Multiline comments with star or without? See license

Comment (MarcusWichelmann) [inline lib/Utils.vala]: I don't know about any styling-rules related to this. The copyright-header is an exception anyway...

File: lib/CMakeLists.txt (hunk 1 above)
Comment (MarcusWichelmann) [inline lib/CMakeLists.txt]: Will change it. ;-)

File: lib/Utils.vala (hunk 2 above)
Comment (Waishon) [inline lib/Utils.vala]: I've used stars 
https://github.com/Waishon/Vala-Lint/blob/MissingBracketsCheck/lib/Checks/MissingBracketsCheck.vala#L50

File: lib/Checks/BlockOpeningBraceSpaceBeforeCheck.vala [hunk 3]
```diff
@@ -38,7 +38,7 @@ public class ValaLint.Checks.BlockOpeningBraceSpaceBeforeCheck : Check {
         char[] chars = line.to_utf8 ();
         bool mistake_found = false;
         for (int i = 0; i < chars.length; i++) {
-            if (i >= 1 && chars[i] == '{' && chars[i - 1] != ' ' && chars[i - 1] != '(') {
+            if (i >= 1 && !Utils.is_char_inside_string (line, i) && chars[i] == '{' && chars[i - 1] != ' ' && chars[i - 1] != '(') {
```
Comment (jeremypw) [inline lib/Checks/BlockOpeningBraceSpaceBeforeCheck.vala]: The number of times the function is called (for error free lines) is proportional to n squared, where n is the length of the line.  Is there a more efficient way of doing this?

**Comment by danrabbit** [comment]
Closing since this seems pretty abandoned

**Pull request closed without merging**